extensions/standard-processors/tests/unit/SplitRecordTests.cpp

```diff
@@ -0,0 +1,96 @@
+    controller_.plan->addController("JsonRecordSetWriter", "JsonRecordSetWriter");
+    proc_->setProperty(processors::SplitRecord::RecordReader, "JsonRecordSetReader");
+    proc_->setProperty(processors::SplitRecord::RecordWriter, "JsonRecordSetWriter");
+  }
+
+  void verifyResults(const ProcessorTriggerResult& results, const std::vector<std::string>& expected_contents) const {
+    REQUIRE(results.at(processors::SplitRecord::Original).size() == 1);
+    REQUIRE(results.at(processors::SplitRecord::Splits).size() == expected_contents.size());
+    auto& split_results = results.at(processors::SplitRecord::Splits);
+    const auto fragment_identifier = split_results[0]->getAttribute("fragment.identifier").value();
+    const auto original_filename = results.at(processors::SplitRecord::Original)[0]->getAttribute("filename").value();
+    for (size_t i = 0; i < expected_contents.size(); ++i) {
+      rapidjson::Document result_document;
+      result_document.Parse(controller_.plan->getContent(split_results[i]).c_str());
+      rapidjson::Document expected_document;
+      expected_document.Parse(expected_contents[i].c_str());
+      CHECK(result_document == expected_document);
+      CHECK(split_results[i]->getAttribute("record.count").value() ==   std::to_string(minifi::utils::string::split(expected_contents[i], "},{").size()));
+      CHECK(split_results[i]->getAttribute("fragment.index").value() == std::to_string(i));
+      CHECK(split_results[i]->getAttribute("fragment.count").value() == std::to_string(expected_contents.size()));
+      CHECK(split_results[i]->getAttribute("fragment.identifier").value() == fragment_identifier);
+      CHECK(split_results[i]->getAttribute("segment.original.filename").value() == original_filename);
+    }
+  }
+
+ protected:
+  std::shared_ptr<core::Processor> proc_ = std::make_shared<processors::SplitRecord>("SplitRecord");
+  SingleProcessorTestController controller_{proc_};
+};
+
+TEST_CASE_METHOD(SplitRecordTestController, "Invalid Records Per Split property", "[splitrecord]") {
+  proc_->setProperty(processors::SplitRecord::RecordsPerSplit, "invalid");
+  auto results = controller_.trigger({InputFlowFileData{"{\"name\": \"John\"}\n{\"name\": \"Jill\"}"}});
+  REQUIRE(results[processors::SplitRecord::Failure].size() == 1);
+  REQUIRE(LogTestController::getInstance().contains("Failed to convert Records Per Split property to an integer", 1s));
+}
+
+TEST_CASE_METHOD(SplitRecordTestController, "Records Per Split property should be greater than zero", "[splitrecord]") {
+  proc_->setProperty(processors::SplitRecord::RecordsPerSplit, "${id}");
+  auto results = controller_.trigger({InputFlowFileData{"{\"name\": \"John\"}\n{\"name\": \"Jill\"}", {{"id", "0"}}}});
+  REQUIRE(results[processors::SplitRecord::Failure].size() == 1);
+  REQUIRE(LogTestController::getInstance().contains("Records per split should be set to a number larger than 0", 1s));
+}
```

Comment:
I think we have similar tests for other processors as well and we link the expression language lib to the test binaries for this specific reason. I think it's worth testing that a specific property supports expression language, and to show it in the test that this is our intention with this property, so that removing the support for it would result in a failure.